File: lib/Pig/Latin.rb
class Pig; end
class Pig::Latin
  def initialize *args
    @english, = args
  end

  def convert
    pig_latin_words = @english.split(/ +/).map do |word|
      lword = word.downcase
      match = lword.match(/^([^aeiou]*)(.*)$/) or
        throw 'error'
      ay = match[1].length == 0 ? 'way' : 'ay'
      pword = match[2] + match[1] + ay
      if word.match /^[A-Z]/
        pword[0].upcase + pword[1..-1]
      else
        pword
      end
    end
    pig_latin_words.join ' '
  end
end

File: lib/Pig/Latin.scala
package Pig

case class Latin(english: String) {
  def convert: String = {
    val pigLatinWords = english.split(" +").map(word => {
      val lword = word.toLowerCase

      val re = "^([^aeiou]*)(.*)$".r
      val match_ = lword match {
        case re(m1, m2) => List(m1, m2)
        case _ => throw new Exception("error")
      }

      val ay = if (match_(0).isEmpty()) "way" else "ay"
      val pword = match_(1) + match_(0) + ay
      if (word(0).isUpper)
        pword.capitalize
      else
        pword
    })
    pigLatinWords mkString " "
  }
}
